### PR TITLE
feature/AT-SCE-UI_Login & Get USERS By Id integration endpoint

### PR DESCRIPTION
**Pull request by Emiliano-Plascencia**
Added POST login operation and get user by id operation in at-sce-api.service.js file.
Added authentication function in at-sce.routes.js file.
The serve.js file imported the paspport file.
In admin.controller.js file signin and logout method was added.

### CODE REVIEW

**Review by irvingesp2190 — APPROVED** [review]

**Review by joseDnim — APPROVED** [review]

File: src/controllers/at-sce.controller.js [hunk 1]
```diff
@@ -20,17 +24,16 @@ atSCEController.renderSigninForm = async (req, res) => {
...
+atSCEController.signin = passport.authenticate("local", {
+    successRedirect: "/calculator",
+    failureRedirect: "/signin",
+    failureFlash: true
+});
 
 // AT-SCE - Logout
 atSCEController.signout = async (req, res) => {
-  console.log("--> atSCEController.signout");
+  req.logout();
```
Comment (at-internship) [inline src/controllers/at-sce.controller.js:35]: can u keep this line `console.log("--> atSCEController.signout");` ?

File: src/services/at-sce-api.service.js [hunk 2]
```diff
@@ -58,8 +77,18 @@ AT_SCE_SERVICE.createUser = (data) => {
   });
 };
 
+// Operation; Create USER - POST /api/v1/users
+AT_SCE_SERVICE.createUser = () => {
```
Comment (at-internship) [inline src/services/at-sce-api.service.js]: this function already exists https://github.com/at-internship/at-sce-ui/blob/develop/src/services/at-sce-api.service.js#L48:L59

File: src/services/at-sce-api.service.js [hunk 3]
```diff
@@ -39,9 +50,17 @@ AT_SCE_SERVICE.getAllUsers = () => {
...
 };
 
-// Operation: Get USER by ID - GET/api/v1/users/{id}
-AT_SCE_SERVICE.getUserById = () => {
-  return {};
+// Operation: Get USERS by ID - GET/api/v1/users/{id}
+AT_SCE_SERVICE.getUsersById = (id) => {
+  return axios({
+    method: "GET",
+    url: `${AT_SCE_SERVICE_URI}/v1/users/` + id,
```
Comment (at-internship) [inline src/services/at-sce-api.service.js]: you can use template literals here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: src/views/partials/navigation.hbs [hunk 4]
```diff
@@ -37,6 +38,11 @@
                         <img src="/img/icons/menu-logout-icon.png" title="Logout" class="icons">
                     </a>
                 </li>
+                {{else}}
+                <li class="nav-item">
+                    <a class="nav-link" href="/users/signin"><b>Login</b></a>
```
Comment (at-internship) [inline src/views/partials/navigation.hbs]: I think shloud be `<a class="nav-link" href="/signin"><b>Login</b></a>`

File: src/services/at-sce-api.service.js [hunk 5]
```diff
@@ -22,9 +24,18 @@ const AT_SCE_SERVICE_URI = process.env.AT_SCE_SERVICE_URI;
...
+AT_SCE_SERVICE.login = (data) => {
+  return axios({
+      method: "POST",
+      url: `${AT_SCE_SERVICE_URI}/v1/login`,
+      data: data,
+      headers: {
+        "content-type": "application/json",
+      },
+    }).catch(function (error) {
+      console.log("Error: " + error.message);
```
Comment (at-internship) [inline src/services/at-sce-api.service.js]: you can use template literals here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: src/services/at-sce-api.service.js [hunk 6]
```diff
@@ -11,6 +11,8 @@
 // Constants
 const axios = require("axios");
 const AT_SCE_SERVICE = {};
+const bcrypt = require("bcryptjs");
```
Comment (at-internship) [inline src/services/at-sce-api.service.js]: do we need this here?

File: src/routes/at-sce.routes.js [hunk 7]
```diff
@@ -12,6 +12,7 @@
 const express = require("express");
 const router = express.Router();
 const path = require("path");
+const helper = require('../helpers/auth.helper')
```
Comment (at-internship) [inline src/routes/at-sce.routes.js]: try this 
`// Helpers
const { isAuthenticated } = require("../helpers/auth.helper");`

File: src/controllers/at-sce.controller.js [hunk 8]
```diff
@@ -9,6 +9,10 @@
  *
  */
 
+//Models
+const passport = require('passport');
+const User = require('../services/at-sce-api.service');
```
Comment (at-internship) [inline src/controllers/at-sce.controller.js]: do we need this here?

File: src/controllers/at-sce.controller.js [hunk 9]
```diff
@@ -9,6 +9,10 @@
  *
  */
 
+//Models
```
Comment (at-internship) [inline src/controllers/at-sce.controller.js]: // This is not a Model

File: src/config/passport.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+            usernameField: "email",
+            passwordField: "password"
+        },
+        async(email, password, done) => {
+            // Match Email's User
+            const request = {
+                email : email,
+                password : password
+            };
+            const user = await sceServiceAPI.login(request);
```
Comment (at-internship) [inline src/config/passport.js]: try this `const userAuth = await sceServiceAPI.login(request);`

File: src/config/passport.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+            
+            if (!user) {
+                return done(null, false, { message: "Not User found." });
+            } else {
+                console.log(user);
```
Comment (at-internship) [inline src/config/passport.js]: here we need a second
try this `const user = await sceServiceAPI.getUserById(userAuth.data);`

**Review by at-internship — CHANGES_REQUESTED** [review]
Please verify code comments

File: src/controllers/at-sce.controller.js (hunk 1 above)
Comment (moselGlez) [reply src/controllers/at-sce.controller.js:35]: sorry i made click, for mistake, in resolve but is not resolve yet

File: src/config/passport.js (hunk 11 above)
Comment (moselGlez) [reply src/config/passport.js]: It doesn't works on my local

**Review by at-internship — APPROVED** [review]
Approved

**Pull request merged**